meshroom/ui/graph.py

```diff
@@ -690,6 +690,29 @@ def moveSelectedNodesBy(self, offset: QPoint):
                 position = Position(node.x + offset.x(), node.y + offset.y())
                 self.moveNode(node, position)
 
+    @Slot()
+    def alignVertically(self):
+        """ All nodes are moved vertically to the same position than the firstly selected node """
+        selectedNodes = self.getSelectedNodes()
+        
+        if len(selectedNodes) < 2:
+            return
+        
+        for selectedNode in selectedNodes:
+            self.moveNode(selectedNode, Position(selectedNode.x, selectedNodes[0].y))
```

Comment:
Need to create an undo group to avoid creating one entry in the undo stack for each node.